py/visualize.py

```diff
@@ -486,48 +475,60 @@ def bubble_chart(responses, poll_ids=None, question_ids=None, xtab1_val="-"):
             x="date",
             y="pct_fav",
             color="country",
-            size=size,
-            size_max=size_max,
             opacity=opacity,
             custom_data=custom_data,
             labels=variable_mapping_inverse,
         )
 
-    # add line for zero net fav
-    fig.add_hline(
-        y=0,
-        line_width=3,
-        line_dash="dot",
-        line_color=GRAY,
-        annotation_text="Zero net favorability",
-        annotation_position="bottom left",
-    )
-
     fig.update_layout(
         clickmode="event+select",
         xaxis=dict(
             title=None,  # date self-explanatory
         ),
         legend_title_text=None,  # country self-explanatory
+        autosize=True,
     )
     fig.update_layout(
         hoverlabel=dict(bgcolor="white", font_size=16, font_family="Arial")
     )
 
     fig.update_traces(
         # Add hover text, which will display instead of custom_data. Custom data is still available to the click event that updates the bar graph
-        # TODO: add plus sign to favorability like in the bar chart
         hovertemplate="<br>".join(
             [
                 "<b>%{customdata[3]}</b>",
                 "%{customdata[2]}",
-                "Net favorability: %{y:+.0f}%",
+                "Net favorability: %{y:+.0f}",
                 "%{customdata[5]}, %{x}",
                 "<extra></extra>",
             ]
         ),
         hoverlabel_align="left",
     )
+    # set size of bubbles
+    fig.update_traces(marker_size=8)
+
+    # add a Range slider to the graph
+    fig.update_xaxes(rangeslider_visible=True)
+
+    # get the date of the most recent poll, and add one month to it
+    date_range_max = (poll_question.date.max() + timedelta(days=30)).strftime(
+        "%Y-%m-%d"
+    )
+
+    initial_range = ["2016-01-01", date_range_max]
+
+    fig["layout"]["xaxis"].update(range=initial_range)
```

Comment:
i think this could also be part of a big `fig.update_layout` call, as `xaxis_range=initial_range`